string address
int32 type
---
bool is_connect false
int32 index 0